holoviews/plotting/bokeh/element.py

```diff
@@ -1881,6 +1885,10 @@ def initialize_plot(self, ranges=None, plot=None, plots=None, source=None):
             if self._subcoord_overlaid:
                 if style_element.label in plot.extra_y_ranges:
                     self.handles['y_range'] = plot.extra_y_ranges.pop(style_element.label)
+
+        if self.apply_hard_bounds:
+            self._apply_hard_bound(element, ranges)
```

Comment:
Will also have to call this on every plot update.

Reply:
done